### ISSUE TITLE
Problem with `eval-with-compile`

### ISSUE THREAD

**Issue by ndw** [ndwarshuis/om.el#6]
I reported [a bug](https://github.com/raxod502/straight.el/issues/462) against `straight.el` that appears to be a problem in `om.el`:

I get:

```
Debugger entered--Lisp error: (void-variable om-elements)
  (-intersection om-branch-nodes-permitting-child-objects om-elements)
  (defconst om-branch-elements-permitting-child-objects (-intersection om-branch-nodes-permitting-child-objects om-elements) ("/Users/ndw/.emacs.d/straight/build/om/om.elc" . 741))
  require(om)
  eval((require (quote om)) nil)
  elisp--eval-last-sexp(t)
  eval-last-sexp(t)
  eval-print-last-sexp(nil)
  funcall-interactively(eval-print-last-sexp nil)
  call-interactively(eval-print-last-sexp nil nil)
  command-execute(eval-print-last-sexp)
```

The maintainer of `straight.el` says:

> That is a bug in the package as far as I can tell. It should not be using eval-when-compile for that code. When I run make compile from the package's Makefile, then manually evaluate the resulting .elc files, I get the same error. Given that, I don't think the problem lies with straight.el.

Hope this is useful.

**Comment by ndwarshuis** [comment]
This should fix it e51aa0b3f4b764aa62fbd7a04e77703a3f6e6710

**Comment by ndw** [comment]
Yep. Thanks!
